airflow-core/src/airflow/api_fastapi/common/exceptions.py

```diff
@@ -79,6 +76,47 @@ def _is_dialect_matched(self, exc: IntegrityError) -> bool:
                 return True
         return False
 
+    def _get_user_friendly_message(self, exc: IntegrityError) -> str:
+       """Convert database error to user-friendly message."""
+       exc_orig_str = str(exc.orig)
+      
+       # Handle DAG run unique constraint
+       if "dag_run.dag_id" in exc_orig_str and "dag_run.run_id" in exc_orig_str:
+           return "A DAG run with this ID already exists. Please use a different run ID."
+          
+       # Handle task instance unique constrain
+       if "task_instance.dag_id" in exc_orig_str and "task_instance.task_id" in exc_orig_str:
+           return "A task instance with this ID already exists. Please use a different task ID."
+          
+       # Handle DAG run logical date unique constraint
+       if "dag_run.dag_id" in exc_orig_str and "dag_run.logical_date" in exc_orig_str:
+           return "A DAG run with this logical date already exists. Please use a different logical date."
+          
+       # Generic unique constraint message
+       return "This operation would create a duplicate entry. Please ensure all unique fields have unique values."
```

Comment:
This seem to handle only duplicated entry errors (unique constraint), while there are many other `IntegrityError` possible.